infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -582,8 +586,28 @@ private boolean happenedAfterWatcherInitialization(PodEvent event)
     containerEventsHandlers.add(handler);
   }
 
-  /** Stops watching the pods inside Kubernetes namespace. */
+  public void watchLogs(PodLogHandler handler, Executor executor) throws InfrastructureException {
```

Comment:
The `executor` is ignored if `logWatcher` already exists, so we're breaking the caller expectations here slightly. It would be good if we could get rid of that parameter and supply it at construction time or somehow made it clear that it might not be used.

Reply:
reworked